airbyte-cdk/bulk/toolkits/load-low-code/src/main/kotlin/io/airbyte/cdk/load/model/spec/Spec.kt

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.model.spec
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.JsonNode
+import io.airbyte.protocol.models.v0.AdvancedAuth
+
+data class Spec(
+    @JsonProperty("connection_specification") val connectionSpecification: JsonNode,
+    @JsonProperty("advanced_auth") val advancedAuth: AdvancedAuth?,
```

Comment:
I'm not 100% comfortable about using the CDK protocol directly there but historically for sources, those have always been aligned 1:1 and if we ever do a breaking change, I assume there will be work here to migrate from `io.airbyte.protocol.models.v0.AdvancedAuth` to `io.airbyte.protocol.models.v1.AdvancedAuth` (or `vX` whatever x value is)

Reply:
For the most part, we shouldn't be referencing protocol directly. As you pointed it, it will be an extra hurdle for versioning. The other one is with speed, we now may have different serialization format. This is why we've been passing interfaces/other impl everywhere for a lot of the protocol objects.

Reply:
> As you pointed it, it will be an extra hurdle for versioning. 

What makes it extra hurdle? I mean, once we migrate connectors to spec v1, there will be work anyway. I don't see this as more work though. 

> The other one is with speed, we now may have different serialization format. 

This is the interface between the manifest and the CDK though. The interface between the connector and the platform is [here in DeclarativeSpecificationFactory](https://github.com/airbytehq/airbyte/blob/maxi297/spec-for-low-code/airbyte-cdk/bulk/toolkits/load-low-code/src/main/kotlin/io/airbyte/cdk/load/spec/DeclarativeSpecificationFactory.kt#L29-L31)